app/models/vm/operations/power.rb

```diff
@@ -18,6 +18,12 @@ module Vm::Operations::Power
       unsupported_reason_add(:start, msg) if msg
     end
 
+    supports :stop do
+      msg = unsupported_reason(:control) unless supports_control?
+      msg ||= _('The VM is not powered on') unless vm_powered_on?
+      unsupported_reason_add(:stop, msg) if msg
+    end
```

Comment:
So this is what we were talking about being an issue with having the `supports` at the base class...now any vm subclass that belongs to a provider and is powered on will say that it `supports :stop`.  Also the class level will say it supports stop.

I see this same pattern for the rest of these operations though, need to investigate if those are handled a different way or if there is a bug here for those.

Reply:
I think they all support the stop operation - so it may be valid at this high level.

Reply:
@agrare the search for `validate_stop` yields [4 results](https://github.com/search?q=org%3AManageIQ+validate_stop&type=code).  Only 1 of them applies to VMs, the other 3 apply to Hosts.  So, I am pretty confident (based upon code inspection) that the VM Stop validation is only defined here.

Reply:
Right the issue is even if all current providers support stop that isn't the same thing as all future subclasses of the Vm model supporting stop (e.g. both IBM VPC and PowerVS providers didn't support stop originally so this was incorrect for a period of time).

The other two here (start and suspend) likely have the same issue since certainly not every Vm type supports suspend.

Reply:
https://github.com/ManageIQ/manageiq/issues/21288 for an example of the issue I'm talking about